File: client/app/services/ocorrenciaService.ts
import api from "../api/apiClient";
import type { Ocorrencia } from "../types";

// Serviço para gerenciar ocorrências/desastres
export const ocorrenciaService = {
  // Listar todas as ocorrências
  listarTodas: async (): Promise<Ocorrencia[]> => {
    const response = await api.get<Ocorrencia[]>("/ocorrencias");
    return response.data;
  },

  // Buscar ocorrência por ID
  buscarPorId: async (id: number): Promise<Ocorrencia> => {
    const response = await api.get<Ocorrencia>(`/ocorrencias/${id}`);
    return response.data;
  },

  // Cadastrar nova ocorrência
  cadastrar: async (ocorrencia: Omit<Ocorrencia, "id">): Promise<Ocorrencia> => {
    const response = await api.post<Ocorrencia>("/ocorrencias", ocorrencia);
    return response.data;
  },

  // Atualizar ocorrência existente
  atualizar: async (id: number, ocorrencia: Partial<Ocorrencia>): Promise<Ocorrencia> => {
    const response = await api.put<Ocorrencia>(`/ocorrencias/${id}`, ocorrencia);
    return response.data;
  },

  // Deletar ocorrência
  deletar: async (id: number): Promise<void> => {
    await api.delete(`/ocorrencias/${id}`);
  },

  // Listar ocorrências recentes (últimas 10)
  listarRecentes: async (): Promise<Ocorrencia[]> => {
    const response = await api.get<Ocorrencia[]>("/ocorrencias/recentes");
    return response.data;
  },
};

File: server/src/main/java/br/com/projeto/service/OcorrenciaService.java
package br.com.projeto.service;

import br.com.projeto.dao.OcorrenciaDao;
import br.com.projeto.model.Ocorrencia;

import java.util.List;

public class OcorrenciaService {

    private final OcorrenciaDao ocorrenciaDao;

    public OcorrenciaService() {
        this.ocorrenciaDao = new OcorrenciaDao();
    }

    public List<Ocorrencia> listarTodas() {
        return ocorrenciaDao.listarTodas();
    }

    public List<Ocorrencia> listarRecentes() {
        return ocorrenciaDao.listarRecentes();
    }

    public Ocorrencia buscarPorId(Long id) {
        return ocorrenciaDao.buscarPorId(id);
    }

    public void inserir(Ocorrencia ocorrencia) {
        ocorrenciaDao.inserir(ocorrencia);
    }

    public void atualizar(Ocorrencia ocorrencia) {
        ocorrenciaDao.atualizar(ocorrencia);
    }

    public void deletar(Long id) {
        ocorrenciaDao.deletar(id);
    }
}